include/openthread/platform/radio.h

```diff
@@ -694,6 +694,22 @@ void otPlatRadioClearSrcMatchShortEntries(otInstance *aInstance);
  */
 void otPlatRadioClearSrcMatchExtEntries(otInstance *aInstance);
 
+/**
+ * Get the radio supported channel mask that the device is allowed to be on.
+ *
+ * @param[in]  aInstance   The OpenThread instance structure.
+ *
+ */
+uint32_t otPlatRadioGetSupportedChannelMask(otInstance *aInstance);
```

Comment:
How about instead of `OPENTHREAD_CONFIG_RADIO_CHANNEL_MASK` we use a radio capability from `otRadioCaps` definition to indicate whether the radio platform supports these APIs or not?

Reply:
otRadioCaps is a run time capacity check, if we use otRadioCaps instead, every platform needs to implement the APIs (real implementation or at least a stub), it make the APIs like a mandatory  feature, but not optional as designed. Do we want to make it a mandatory feature?

Reply:
Good point on radio capability. 

I was also thinking of adding weak implementation (noticed @bukepo suggested the same idea). 
- We can add `OT_TOOL_WEAK` default implementation for both `otPlatRadioGetSupportedChannekMask` and `otPlatRadioGetPreferredChannelMask()` and radio platform code can choose to override either definition and provide its own implementation. 
  - We can make default implementation of `otPlatRadioGetSupportedChannekMask()` to return `Phy::kSupportedChannels`
  - And default implementation of `otPlatRadioGetPreferredChannelMask()` actually returning the `otPlatRadioGetSupportedChannelMask()` (this way radio platform can choose to just override the `GetSupportedChannelMask()`)

- I feel a weak implementation of the API may be simpler and cleaner (no need for a new feature `OPENTHREAD_CONFIG_RADIO_CHANNEL_MASK` and/or the spinel capability). 
- I understand your point  that having `OPENTHREAD_CONFIG_RADIO_CHANNEL_MASK` would reduce code size but this feature is not adding a lot of code and is not that big, so may be fine to have it be available on all builds. I perosnally think simplicity of design may be more preferable in this case over the small code size saving.

Thoughts?

Reply:
Thanks for your suggestions, it's ok for me to remove the macro definition with little code size increasement, and agree that weak method is much simpler. The PR is updated accordingly.